converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/ResourceId/ServerResourceId.cs

```diff
@@ -18,18 +19,42 @@ public class ServerResourceId : IResourceId, IEquatable<ServerResourceId>
         private ResourceReference _resourceReference;
 
         public ServerResourceId(ResourceType resourceType, string resourceId)
+            : this(EnumUtility.GetLiteral(resourceType), resourceId)
         {
-            EnsureArg.EnumIsDefined(resourceType, nameof(resourceType));
+        }
+
+        public ServerResourceId(string typeName, string resourceId)
+        {
+            EnsureArg.IsNotNullOrWhiteSpace(typeName, nameof(typeName));
             EnsureArg.IsNotNullOrWhiteSpace(resourceId, nameof(resourceId));
 
-            ResourceType = resourceType;
+            TypeName = typeName;
             ResourceId = resourceId;
         }
 
+        /// <summary>
+        /// Gets the type name.
+        /// </summary>
+        public string TypeName { get; }
+
         /// <summary>
         /// Gets the resource type.
         /// </summary>
-        public ResourceType ResourceType { get; }
+        [Obsolete("Please use TypeName instead.")]
+        public ResourceType ResourceType
```

Comment:
@abdullah248 / @pengchen0692 - The big change that impacts us is the removal of `ResourceType` from `Resource` and has instead been replaced by the more generic `TypeName`. Should we continue to expose `ResourceType` here or replace it?

Total removal is a breaking change, so in this PR interation I have obsoleted it.

Reply:
I think obsoleting it for now is fine.